src/backend/distributed/deparser/qualify_statistics_stmt.c

```diff
@@ -83,6 +86,12 @@ QualifyDropStatisticsStmt(Node *node)
 
 			if (OidIsValid(statsOid))
 			{
+				/* user should own the statistic object */
```

Comment:
This is way too low in the call stack to check the owner of the object. Is it possible to move this check to earlier?